From f23636ad25f5c40cdf00371352c2e4c2d4fe4c88 Mon Sep 17 00:00:00 2001
From: Cyprian Lech <clech@plume.com>
Date: Thu, 31 Mar 2022 07:34:05 +0000
Subject: [PATCH] RDKB-41318: Add support for opensync.service

Reason for change: Better integration OpenSync into Systemd
Test Procedure: Check if OpenSync is running correctly
Risks: Low

Change-Id: I8c1a8d19882eb932904627f907628d1461eb2cb9
Signed-off-by: Cyprian Lech <clech@plume.com>
---

diff --git a/scripts/handle_mesh b/scripts/handle_mesh
index 6524256..408a8d4 100644
--- a/scripts/handle_mesh
+++ b/scripts/handle_mesh
@@ -57,18 +57,24 @@
     return 1
 }
 
+is_opensync_service()
+{
+    systemctl list-unit-files opensync.service | grep -q opensync.service
+}
+
 if [ $# -eq 0 ]; then
  echo "No arguments passed"
  exit 0
 else
  platform_checks
  if [ "$OPENSYNC_ENABLE" == "true" ];then
-    if [ -z "$(pidof ovsdb-server)" ]
+    if [ -z "$(pidof ovsdb-server)" ] || is_opensync_name
     then
-        /usr/opensync/scripts/managers.init $1
-    elif is_opensync_name
-    then
-        /usr/opensync/scripts/managers.init $1
+        if is_opensync_service; then
+            systemctl $1 opensync.service
+        else
+            /usr/opensync/scripts/managers.init $1
+        fi
     else
         echo "Opensync will be effective only after reboot"
         /usr/plume/scripts/managers.init $1
